.jenkins/Jenkinsfile

```diff
@@ -1,5 +1,8 @@
 pipeline {
   agent any
+  options {
+    timeout(time: 30, unit: 'MINUTES')
```

Comment:
We might want to add a timeout to each individual stage, and not the overarching stage, since our more in-demand systems (ACC VMs) can get overutilized and make an entire pipeline run take more than 30 minutes, even if each job only takes 7 minutes or so.

Reply:
So, it doesn't appear we need to do this. I did this based off the [buildTimeTrend](https://oe-jenkins.eastus.cloudapp.azure.com/job/Bors/job/staging/buildTimeTrend) and while there are occasional instances of 30 minutes being exceeded, I think we should consider that a failure.